content/5_Plat/Convex_Hull.mdx

```diff
@@ -85,22 +84,197 @@ export const problems = {
+
+<Spoiler title="Convex Hull Visualization">
+
+![Convex Hull Visualization](./assets/convexhull.jpg)
+
+</Spoiler>
+
 <FocusProblem problem={problems.sample[0]} />
 
 ## With Graham Scan
 
-- [cp-algo](https://cp-algorithms.com/geometry/grahams-scan-convex-hull.html)
-- [My Implementation](<https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h>)
+<Resources>
+	<Resource source="Wikipedia" title="Graham Scan" url="https://en.wikipedia.org/wiki/Graham_scan" />
+	<Resource source="VisuAlgo" title="Graham Scan Visualization" url="https://visualgo.net/en/convexhull" />
+	<Resource source="UCSD" title="Original Graham Scan Paper" url="http://www.math.ucsd.edu/~ronspubs/72_10_convex_hull.pdf" />
+</Resources>
+
+<Youtube id="9rQMLpQn5xQ"/>
 
 ### Solution
 
-<IncompleteSection />
+<Resources>
+	<Resource source="Benq" title="Graham Scan Implementation" url="https://github.com/bqi343/USACO/blob/master/Implementations/content/geometry%20(13)/Polygons/ConvexHull2.h" />
+</Resources>
+
+<LanguageSection>
+
+<CPPSection>
+
+The Graham Scan algorithm works in 3 steps. First, it sorts all of the $n$ points by their counterclockwise angle around a pivot $P_0$, breaking ties by distance. This algorithm uses the leftmost (and bottommost if there is a tie), point as $P_0$. 
+
+We maintain a stack containing the points of the convex hull, such that the set of points in the stack is convex. As a result, we only have to check the candidate point against the topmost 2 points.
+
+To start the creation of the convex hull, we choose 2 points. The pivot (first point), and the second point based on our initial sorting.
+
+In order for the hull to be convex, every three consecutive points of the stack must contain a counterclockwise turn.
+
+Denote our stack as $\texttt{C}$, the top element of $\texttt{C}$ as $\texttt{C}[i]$ and $\texttt{cand}[j]$ as the $j$-th sorted point.
+
+There are 2 cases for our 3 points: either the turn from $\texttt{C}[i-1] \rightarrow \texttt{C}[i] \rightarrow \texttt{cand}[j]$ is counterclockwise or not. If the turn is not counterclockwise, then we know that our current set of points cannot be convex, and thus we pop the top element off $\texttt{C}$ and proceed to check convexity of the point until the turn is counterclockwise. Otherwise the turn is counterclockwise, and thus we include it in our hull.
+
+To understand why, consider those three points: the top two points in the stack, and our candidate point. Since our turn is counterclockwise, we are looking for such points that make a counterclockwise turn. It is easy to see that when a clockwise turn it is made, the polygon can no longer be convex.
+
+<Spoiler title="Illustration">
+
+![Illustration of Convex Hull](./assets/convexhullpts.jpg)
+
+</Spoiler>
+
+<Spoiler title="Worked Example">
+
+> Find the convex hull of the points $(1,2),(2,3),(5,3),(3,2),(2,0),(6,2)$.
+
+[Animated](https://convexhull.tiiny.site/)
```

Comment:
I prefer the method of animation used in the video:

https://youtu.be/hAZ3GaEzA04?t=692

related code can be found here: https://lukakalinovcic.github.io/geodeb/gallery/hull.html

(though the rainbow is distracting)

(also, you should link to geodeb)